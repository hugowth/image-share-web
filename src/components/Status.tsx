import React from "react";
import { Link, navigate } from "gatsby";
import { isLoggedIn, logout } from "../helpers/auth";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(() => ({
  status: {
    background: "lightgrey",
    fontSize: "87.5%",
    padding: "0.25rem",
  },
  status__text: {
    margin: "0 auto",
    maxWidth: 640,
    textAlign: "right",
  },
}));

const Status = () => {
  const classes = useStyles();

  let details;
  if (!isLoggedIn()) {
    details = (
      <p className={classes.status__text}>
        To get the full app experience, youâ€™ll need to
        {` `}
        <Link to="/app/login">log in</Link>.
      </p>
    );
  } else {
    details = (
      <p className={classes.status__text}>
        Logged!
        <a
          href="/"
          onClick={(event) => {
            event.preventDefault();
            logout();
          }}
        >
          log out
        </a>
      </p>
    );
  }

  return <div className={classes.status}>{details}</div>;
};

export default Status;
