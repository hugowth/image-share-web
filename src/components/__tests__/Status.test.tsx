import React from "react";
import renderer from "react-test-renderer";

import Status from "../Status";

describe("Status", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Status />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
