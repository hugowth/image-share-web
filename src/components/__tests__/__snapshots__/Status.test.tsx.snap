// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status renders correctly 1`] = `
<div
  className="makeStyles-status-1"
>
  <p
    className="makeStyles-status__text-2"
  >
    To get the full app experience, youâ€™ll need to
     
    <a
      href="/app/login"
    >
      log in
    </a>
    .
  </p>
</div>
`;
